### PR TITLE
Sovrn Adapter accept adUnitCodes

### PR DESCRIPTION
**Pull request by ikagotso**

### CODE REVIEW

**Comment by cpabst** [comment]
There seems to be something wrong with this PR now. 252 files?

**Comment by ikagotso** [comment]
> There seems to be something wrong with this PR now. 252 files?

It was a temporary issue. Should be normal now

File: src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java [hunk 1]
```diff
@@ -114,6 +118,14 @@ private String resolveTagId(ExtImpSovrn sovrnExt) {
         return tagId;
     }
 
+    private ObjectNode resolveImpExt(ExtImpSovrn sovrnExt, ObjectNode impExt) {
+        final ObjectNode sovrnImpExt = impExt.deepCopy();
+        sovrnImpExt.remove(EXT_BIDDER_PARAM);
```
Comment (cpabst) [inline src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java:123]: Why are we removing `bidder` here?

File: src/main/java/org/prebid/server/proto/openrtb/ext/request/sovrn/ExtImpSovrn.java [hunk 2]
```diff
@@ -17,4 +17,7 @@ public class ExtImpSovrn {
     String legacyTagId;
 
     BigDecimal bidfloor;
+
+    @JsonProperty("adUnitCode")
```
Comment (cpabst) [inline src/main/java/org/prebid/server/proto/openrtb/ext/request/sovrn/ExtImpSovrn.java:21]: Isn't this redundant?

**Review by cpabst — COMMENTED** [review]
A couple of questions

File: src/test/resources/org/prebid/server/it/openrtb2/sovrn/test-sovrn-bid-request.json [hunk 3]
```diff
@@ -20,9 +20,7 @@
       "tagid": "tag_id",
       "ext": {
         "tid": "${json-unit.any-string}",
-        "bidder": {
-          "tagid": "tag_id"
-        }
```
Comment (sejanick) [inline src/test/resources/org/prebid/server/it/openrtb2/sovrn/test-sovrn-bid-request.json:25]: Was there a reason we removed this bidder/tagid value in this PR?

Comment (ikagotso) [reply src/test/resources/org/prebid/server/it/openrtb2/sovrn/test-sovrn-bid-request.json:25]: I tried comparing it to Blackbird's openrtb-sovrn.proto: 

[](https://github.com/sovrn/blackbird/blob/23bc2e065166a5c705d9b1f7d47fc03dbb7b59d3/blackbird-openrtb-ext/src/main/proto/openrtb-sovrn.proto#L26C1-L34C1)

there is no bidder field handled there, so I thought it was redundant. Please correct me if I am wrong.

File: src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java (hunk 1 above)
Comment (ikagotso) [reply src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java:123]: I tried comparing it to Blackbird's openrtb-sovrn.proto:

https://github.com/sovrn/blackbird/blob/23bc2e065166a5c705d9b1f7d47fc03dbb7b59d3/blackbird-openrtb-ext/src/main/proto/openrtb-sovrn.proto#L26C1-L34C1

there is no bidder field handled there, so I thought it was redundant. Please correct me if I am wrong.

**Comment by cpabst** [comment]
@ikagotso I need to check the Go code. I'll let you know when I have a better answer.

**Review by sejanick — APPROVED** [review]
This LGTM, as far as I can tell.  Thank you @cpabst  for double-checking things as well.

**Comment by ikagotso** [comment]
> @ikagotso I need to check the Go code. I'll let you know when I have a better answer.

@cpabst Have you checked the Go adapter by any chance?

**Comment by cpabst** [comment]
@ikagotso No, I haven't. I'll do that now.

File: src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java (hunk 1 above)
Comment (cpabst) [reply src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java:123]: OK, I've figured out what was bothering me. The ExtImpSovrn class has nothing to do with our protobuf definition in Blackbird It is a local class in prebid server (org/prebid/server/proto/openrtb/ext/request/sovrn/ExtImpSovrn.java). Here's an example of the javascript that is passing into the adapter.
https://docs.prebid.org/dev-docs/bidders/sovrn.html
I believe the line `sovrnImpExt.remove(EXT_BIDDER_PARAM);` actually does nothing.

Comment (cpabst) [reply src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java:123]: Do tests pass with your changes?

**Review by cpabst — COMMENTED** [review]
Finally figured out what I wanted to say. 😄

File: src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java (hunk 1 above)
Comment (ikagotso) [reply src/main/java/org/prebid/server/bidder/sovrn/SovrnBidder.java:123]: Of course tests pass with my changes.
Yes ExtImpSovrn is local class in prebid server, but eventually the Bidrequest generated in prebid-server gets to prebid endpoint in Blackbird and gets parsed according to proto, doesn't it?

**Review by cpabst — APPROVED** [review]
LGTM

**Review by sejanick — APPROVED** [review]
LGTM, thanks Irakli!

**Comment by ikagotso** [comment]
Merged on original prebid-server-java

**Pull request closed without merging**